core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -750,7 +749,12 @@ abstract static class Unix extends Local {
         @CheckForNull
         @Override
         public OSProcess get(@NonNull Process proc) {
-            return get(Ints.checkedCast(proc.pid()));
+            long longPid = proc.pid();
+            int intPid = (int) longPid;
+            if (intPid != longPid) {
+              throw new IllegalArgumentException("Out of range: " + longPid);
+            }
+            return get(intPid);
```

Comment:
Or
```suggestion
            return get(Math.toIntExact(proc.pid()));
```
throwing `ArithmeticException("integer overflow")` in case of overflow.